examples/standalone/acoustic_comms_demo/acoustic_comms_demo.cc

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright (C) 2022 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+/*
+ * Check README for detailed instructions.
+ * Usage:
+ *   $ gz sim -r worlds/acoustic_comms_demo.sdf
+ *   $ acoustic_comms_demo
+ */
+
+#include <atomic>
+#include <chrono>
+#include <string>
+#include <thread>
+#include <vector>
+
+#include <gz/msgs.hh>
+#include <gz/transport.hh>
+
+using namespace gz;
+
+std::atomic<bool> tethysMsgReceived = false;
+std::atomic<bool> daphneMsgReceived = false;
+
+void cbTethys(const msgs::Dataframe &_msg)
+{
+  tethysMsgReceived = true;
+}
+
+void cbDaphne(const msgs::Dataframe &_msg)
+{
+  daphneMsgReceived = true;
+}
+
+int main(int argc, char** argv)
+{
+  std::vector<std::string> ns;
+  ns.push_back("triton");
+  ns.push_back("tethys");
+  ns.push_back("daphne");
+
+  transport::Node node;
+
+  std::vector<std::string> rudderTopics;
+  rudderTopics.resize(ns.size(), "");
+  std::vector<gz::transport::Node::Publisher> rudderPubs;
+  rudderPubs.resize(ns.size());
+
+  std::vector<std::string> propellerTopics;
+  propellerTopics.resize(ns.size(), "");
+  std::vector<gz::transport::Node::Publisher> propellerPubs;
+  propellerPubs.resize(ns.size());
+
+  // Set up topic names and publishers
+  for (int i = 0; i < ns.size(); i++)
+  {
+    propellerTopics[i] = gz::transport::TopicUtils::AsValidTopic(
+      "/model/" + ns[i] + "/joint/propeller_joint/cmd_pos");
+    propellerPubs[i] = node.Advertise<gz::msgs::Double>(
+      propellerTopics[i]);
+  }
+
+  std::vector<double> propellerCmds = {-20, 0, 0};
+
+  // Setup publishers and callbacks for comms topics.
+  auto senderAddressTriton = "1";
+
+  auto receiverAddressTethys = "2";
+  auto receiverAddressDaphne = "3";
```

Comment:
@adityapande-1995 nit: why use `std::vector` for some variables (for pubs, for topics) and explicit, vehicle specific names for others (for addresses, for messages)? Why not use only names or only indices?

Reply:
Fixed https://github.com/gazebosim/gz-sim/pull/1693/commits/387a98b0e97bb51779c7f09bcae3f00564678959